### PR TITLE
[IMP] project: stage in project task sharing should also display   duration

### PR DESCRIPTION
**Pull request by daiduongnguyen-odoo**
-In the backend view, internal user can view the duration on each stage
but in the form view of project sharing, portal user only see stage and
can't see the duration between them. This commit apply the duration for
portal user view

-The 2nd commit change statusbar to statusbar_duration for stage of project
in case we have enabled it

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by duong77476-viindoo** [comment]
Hi @adrienmilis @luve-odoo Can this consider merge in stable v17, if ok can you review please thank you

**Comment by duong77476-viindoo** [comment]
Hello @tde-banana-odoo who can i ask for this Proposal ?

**Comment by tde-banana-odoo** [comment]
@duong77476-viindoo This cannot be done in stable as it alters db scheme. And concerning the feature itself I let @xavierbol decide :) .

**Comment by xavierbol** [comment]
Hello @duong77476-viindoo 

Sorry for the delay! 🙈 

I agree with Thibault, we cannot do that in stable.

However, we can indeed improve the current behavior to also display the duration in project sharing and enable the feature for the project stages. 🙂 

Can you retarget your work in master? 🙂 

Thanks in advance! 🙂

**Comment by duong77476-viindoo** [comment]
@xavierbol Thank you for the quick replay, i'll do that after Vietnamese holidays hehe, thank again

**Comment by xavierbol** [comment]
@duong77476-viindoo okay no problem. 🙂 

Enjoy your holiday! 🙂

**Comment by duong77476-viindoo** [comment]
@xavierbol I have changed the target branch to master

**Review by xavierbol — COMMENTED** [review]
Sorry for the delay, I was off and a bit busy these days.

I tested your branch, functionally everything works as expected, thank you very much! 😁 

I left 2 small comments after that it will look good to me. 🙂

File: addons/project/models/project_project.py [hunk 1]
```diff
@@ -17,10 +17,18 @@
...
     _name = "project.project"
     _description = "Project"
-    _inherit = ['portal.mixin', 'mail.alias.mixin', 'rating.parent.mixin', 'mail.thread', 'mail.activity.mixin']
+    _inherit = [
+        'portal.mixin',
+        'mail.alias.mixin',
+        'rating.parent.mixin',
+        'mail.thread',
+        'mail.activity.mixin',
+        'mail.tracking.duration.mixin'
```
[the remark this answers is not in the document]
Comment (duong77476-viindoo) [reply addons/project/models/project_project.py]: Nice this is the best explanation for me so far keke
Done

File: addons/project/models/project_task.py [hunk 2]
```diff
@@ -51,6 +51,7 @@
     'repeat_until',
     'recurrence_id',
     'recurring_count',
+    'duration_tracking'
```
[the remark this answers is not in the document]
Comment (duong77476-viindoo) [reply addons/project/models/project_task.py]: Done

**Comment by xavierbol** [comment]
LGTM thanks again for your work! 💪 🚀 

robodoo r+

**Comment by xavierbol** [comment]
robodoo rebase-ff

**Pull request closed without merging**